### PR TITLE
Immutable data deserialization support

### PR DESCRIPTION
**Pull request by japplegame**
Adds the ability to deserialize immutable data. See unittest.

### CODE REVIEW

File: source/vibe/data/serialization.d [hunk 1]
```diff
@@ -446,8 +446,12 @@ private void serializeImpl(Serializer, alias Policy, T, ATTRIBUTES...)(ref Seria
 	} else static assert(false, "Unsupported serialization type: " ~ T.stringof);
 }
 
+private T deserializeImpl(T, alias Policy, Serializer, ATTRIBUTES...)(ref Serializer deserializer) if(!isMutable!T)
+{
+	return cast(T) deserializeImpl!(Unqual!T, Policy, Serializer, ATTRIBUTES)(deserializer);
```
Comment (Geod24) [inline source/vibe/data/serialization.d]: I'd use `assumeUnique` here.

**Comment by japplegame** [comment]
In this case assumeUnique will not help, because it works only with arrays.

File: source/vibe/data/serialization.d (hunk 1 above)
Comment (japplegame) [inline source/vibe/data/serialization.d]: In this case assumeUnique will not help, because it works only with arrays.

Comment (Geod24) [inline source/vibe/data/serialization.d]: Wut. I always assumed it worked with everything and was just documenting.
I'm not familiar with this part, but could you also add an unittest for a `const`/`immutable` object if there's not one already ?

Comment (japplegame) [inline source/vibe/data/serialization.d]: I slightly modified the unittest.

**Pull request merged**